src/vsr/replica.zig

```diff
@@ -3338,6 +3416,19 @@ pub fn Replica(
             return true;
         }
 
+        /// Returns the highest op that can be received, considering the current checkpoint.
+        fn op_ceiling(self: *const Self) u64 {
+            assert(self.op_checkpoint <= self.commit_min);
+            assert(self.op_checkpoint <= self.op);
+            assert((self.op_checkpoint + 1) % config.lsm_batch_multiple == 0);
+
+            if (self.op_checkpoint == 0) {
+                return config.journal_slot_count - 1;
+            } else {
+                return self.op_checkpoint + config.journal_slot_count;
```

Comment:
I think there may be an off-by-one here, since `op_checkpoint` is an index and `journal_slot_count` is a count.

Adding them together produces a count, not an index. To get the index, we need to subtract 1.

I think this also means we can get rid of the branch here, since it's then the same calculation for both branches.